packages/lesswrong/lib/vulcan-core/default_resolvers.ts

```diff
@@ -3,38 +3,93 @@ import { restrictViewableFieldsMultiple, restrictViewableFieldsSingle } from '..
 
 interface DefaultResolverOptions {
   cacheMaxAge: number
 }
 
+const logMissingFragmentNames = false;
+
+const maxAllowedSkip = 2000;
+
 const defaultOptions: DefaultResolverOptions = {
   cacheMaxAge: 300,
 };
 
+const getFragmentNameFromInfo = (
+  {fieldName, fieldNodes}: GraphQLResolveInfo,
+  resultFieldName: string,
+): string | null => {
+  const query = fieldNodes.find(
+    ({name: {value}}) => value === fieldName,
+  );
+  const mainSelections = query?.selectionSet?.selections;
+  const results = mainSelections?.find(
+    (node) => node.kind === "Field" && node.name.value === resultFieldName,
+  ) as FieldNode | undefined;
+  const resultSelections = results?.selectionSet?.selections;
+  const fragmentSpread = resultSelections?.find(
+    ({kind}) => kind === "FragmentSpread",
+  ) as FragmentSpreadNode | undefined;
+  const fragmentName = fragmentSpread?.name.value;
+  if (!fragmentName) {
+    if (logMissingFragmentNames) {
+      const data = JSON.stringify(fieldNodes, null, 2);
+      // eslint-disable-next-line no-console
+      console.error("Fragment name not found for", fieldName, data);
+    }
+    return null;
+  }
+  return fragmentName;
+}
+
 // Default resolvers. Provides `single` and `multi` resolvers, which power the
 // useSingle and useMulti hooks.
-export function getDefaultResolvers<N extends CollectionNameString>(collectionName: N, options?: Partial<DefaultResolverOptions>) {
+export function getDefaultResolvers<N extends CollectionNameString>(
+  collectionName: N,
+  options?: Partial<DefaultResolverOptions>,
+) {
   type T = ObjectsByCollectionName[N]
   const typeName = getTypeName(collectionName);
   const resolverOptions = {...defaultOptions, options};
-  
+
   return {
     // resolver for returning a list of documents based on a set of query terms
 
     multi: {
       description: `A list of ${typeName} documents matching a set of query terms`,
 
-      async resolver(root: void, args: { input: {terms: ViewTermsBase, enableCache?: boolean, enableTotal?: boolean, createIfMissing?: Partial<T>} }, context: ResolverContext, { cacheControl }: AnyBecauseTodo) {
-        const startResolve = Date.now()
+      async resolver(
+        root: void,
+        args: {
+          input: {
+            terms: ViewTermsBase & Record<string, unknown>,
+            enableCache?: boolean,
+            enableTotal?: boolean,
+            createIfMissing?: Partial<T>,
+          },
+        },
+        context: ResolverContext,
+        info: GraphQLResolveInfo,
+      ) {
+        // const startResolve = Date.now()
         const input = args?.input || {};
         const { terms={}, enableCache = false, enableTotal = false } = input;
         const logger = loggerConstructor(`views-${collectionName.toLowerCase()}-${terms.view?.toLowerCase() ?? 'default'}`)
         logger('multi resolver()')
         logger('multi terms', terms)
 
+        // Don't allow API requests with an offset provided >2000. This prevents some
```

Comment:
Currently, all queries are run through `performQueryFromViewParameters` which performs this check. I moved it here to make sure it is still used when we use the SQL fragments code path which bypasses `performQueryFromViewParameters`.